drip/views.py

```diff
@@ -81,3 +82,39 @@ def post(self, *args: Any, **kwargs: Any):
             self.campaign.unsubscribed_users.add(self.user.pk)
             self.post_sucess = True
         return self.render_to_response(context)
+
+
+class UnsubscribeView(TemplateView):
+    template_name = "unsubscribe_general.html"
+    invalid_template_name = "unsubscribe_general_invalid.html"
+    success_template_name = "unsubscribe_general_success.html"
+
+    def _set_user(self, **kwargs: Any):
+        uidb64 = kwargs.get("uidb64", "")
+        token = kwargs.get("token", "")
+        self.user = EmailToken.validate_user_uidb64_token(uidb64, token)
+        self.post_sucess = False
+
+    def dispatch(self, *args: Any, **kwargs: Any):
+        self._set_user(**kwargs)
+        return super().dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+        context_data = super().get_context_data(**kwargs)
+        context_data["user"] = self.user
+        return context_data
+
+    def get_template_names(self) -> List[str]:
+        template_names = super().get_template_names()
+        if not self.user:
+            template_names = [self.invalid_template_name]
+        if self.user and self.post_sucess:
+            template_names = [self.success_template_name]
+        return template_names
```

Comment:
```suggestion
        if not self.user or (self.user and self.post_sucess):
            return [self.invalid_template_name]
        return super().get_template_names()
```

Reply:
Just an optional suggestion

Reply:
I like it, I just apply this suggestion!